### PR TITLE
Migrate issue template from markdown to yaml

### PR DESCRIPTION
**Pull request by zhongjiajie**
Change issue template to yaml make our issue form
more user friendly, and also make our issue structured.
It have some rule to make sure user have to fill
before they submit issue.

See more detail in
https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      label: Configuration
+      description: Include your SQLFluff configuration here
+    validations:
+      required: true
+
+  - type: checkboxes
+    attributes:
+      label: Are you willing to submit PR?
+      description: >
+        This is absolutely not required, but we are happy to guide you in the contribution process
```
Comment (barrywhart) [inline .github/ISSUE_TEMPLATE/bug-report.yml:85]:
```suggestion
        This is absolutely not required, but we are happy to guide you in the contribution process,
```

File: .github/ISSUE_TEMPLATE/documentation.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+body:
+  - type: checkboxes
+    attributes:
+      label: Search before asking
+      description: >
+        Please make sure to search in the [issues](https://github.com/sqlfluff/sqlfluff/issues) first to see
+        whether the same issue was reported already.
+      options:
+        - label: >
+            I had searched in the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found
```
Comment (barrywhart) [inline .github/ISSUE_TEMPLATE/documentation.yml:14]: Same suggestion as for the bug template.

File: .github/ISSUE_TEMPLATE/documentation.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  - type: textarea
+    attributes:
+      label: Issue/Suggested Improvement
+      description: The issue you found or the improvement you suggested.
+    validations:
+      required: true
+
+  - type: checkboxes
+    attributes:
+      label: Are you willing to submit PR?
```
Comment (barrywhart) [inline .github/ISSUE_TEMPLATE/documentation.yml:35]: Same suggestion as for the bug template.

File: .github/ISSUE_TEMPLATE/enhancement.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      options:
+        - label: >
+            I had searched in the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found
+            no similar issues.
+          required: true
+
+  - type: textarea
+    attributes:
+      label: Description
+      description: A short description of your feature
```
Comment (barrywhart) [inline .github/ISSUE_TEMPLATE/enhancement.yml:21]:
```suggestion
      description: A short description of the feature
```

File: .github/ISSUE_TEMPLATE/enhancement.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: Use case
+      description: What do you want to happen?
+      placeholder: >
+        Rather than telling us how you might implement this feature, try to take a
+        step back and describe what you are trying to achieve.
```
Comment (barrywhart) [inline .github/ISSUE_TEMPLATE/enhancement.yml:31]: 💯

File: .github/ISSUE_TEMPLATE/enhancement.yml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+  - type: textarea
+    attributes:
+      label: Dialect
+      description: What dialect you expect to add this feature?
```
Comment (barrywhart) [inline .github/ISSUE_TEMPLATE/enhancement.yml:36]:
```suggestion
      description: If the enhancement relates to a particular dialect, which one?
```

File: .github/ISSUE_TEMPLATE/enhancement.yml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+  - type: checkboxes
+    attributes:
+      label: Are you willing to submit PR?
+      description: >
+        This is absolutely not required, but we are happy to guide you in the contribution process
+        especially if you already have a good understanding of how to implement the fix.
+        SQLFluff is a totally community-driven project and we love to bring new contributors in.
+      options:
+        - label: Yes I am willing to submit a PR!
```
Comment (barrywhart) [inline .github/ISSUE_TEMPLATE/enhancement.yml:48]: Same suggestion as for the bug template.

**Review by barrywhart — DISMISSED** [review]
This looks nice! I didn't know GitHub supported this.

I'd like @tunetheweb to review this as well, since he has worked a lot in this area and is often the first person to look the issues list.

**Comment by zhongjiajie** [comment]
> 

FYI @barrywhart ,  I make some changed base on you suggstion in 23faff7, I think is ready to go, please take a look @tunetheweb

**Review by alanmcruickshank — APPROVED** [review]
I don't think there's a good way of testing these in advance - but these look like a great improvement and I'm happy to iterate on from here 👍

**Comment by tunetheweb** [comment]
I was going to have a look at it in my fork. Gimme an hour or two to do that.

**Comment by zhongjiajie** [comment]
> I add example issue chooser in https://github.com/zhongjiajie/sqlfluff/issues/new/choose, PTAL @alanmcruickshank @tunetheweb @barrywhart

@tunetheweb I have a test in my own fork, maybe you could take a look on it.

**Review by tunetheweb — CHANGES_REQUESTED** [review]
This is excellent. Never knew about this functionality in GitHub.

I do have a few suggestions though.

**Comment by zhongjiajie** [comment]
Hi @tunetheweb @barrywhart @alanmcruickshank , I upgrade this PR right now. PTAL

**Review by tunetheweb — APPROVED** [review]
LGTM!

**Comment by zhongjiajie** [comment]
Thanks~

**Comment by tunetheweb** [comment]
Thank you!

Can we do same for PRs?

**Comment by zhongjiajie** [comment]
> Can we do same for PRs?

No, we can not, GitHub do not support this, at least for now

**Comment by tunetheweb** [comment]
> No, we can not, GitHub do not support this, at least for now

😞 Hopefully soon! Would be very useful there too.

**Comment by zhongjiajie** [comment]
I check our issue creator, it looks great now 👍 
image

**Pull request merged**